### PR TITLE
sx_node_galerkin: fix access to uninitialized memory

### PR DESCRIPTION
**Pull request by lucafedeli88**
This PR fixes a warning stating that `sx_node_galerkin` could be used before initialization.

### CODE REVIEW

File: Source/Particles/Gather/FieldGather.H
Comment (dpgrote) [inline Source/Particles/Gather/FieldGather.H]: Thanks for the fix! I had noticed this warning also. An alternate method is to do

```
amrex::Real sx_node_galerkin[depos_order + 1 - galerkin_interpolation] = {0._rt};
amrex::Real sx_cell_galerkin[depos_order + 1 - galerkin_interpolation] = {0._rt};
```

Comment (ax3l) [reply Source/Particles/Gather/FieldGather.H]: Yep, that's a lesser known feature of array initialization in C++, just for reference:
https://en.cppreference.com/w/c/language/array_initialization

**Review by ax3l — APPROVED** [review]
Thank you for fixing this.
I'll leave it to you and @dpgrote to decide if you want to use array initialization instead.

**Comment by ax3l** [comment]
Let me also mark this a bug, because it likely is one and causes undefined behavior.

File: Source/Particles/Gather/FieldGather.H
Comment (lucafedeli88) [reply Source/Particles/Gather/FieldGather.H]: Its' much better like this! I didn't know that it was possible to initialize C-style arrays in this way. @dpgrote, if you want to "insert a suggestion" I will accept your commit (otherwise, I can do that).

Comment (dpgrote) [reply Source/Particles/Gather/FieldGather.H]: @lucafedeli88 You can make the change.

Comment (lucafedeli88) [reply Source/Particles/Gather/FieldGather.H]: done ! As soon as tests pass I will merge

**Review by dpgrote — APPROVED** [review]
This looks good. Thanks for making the fix!

**Review by RemiLehe — APPROVED** [review]

**Pull request merged**
